src/ray/raylet/worker_pool.cc

```diff
@@ -62,6 +62,19 @@ bool RemoveWorker(
     const std::shared_ptr<ray::raylet::WorkerInterface> &worker) {
   return worker_pool.erase(worker) > 0;
 }
+
+// If both `ask` and `have` are set, they must match. If either of them is not set, it
+// is considered a match.
+bool OptionalMatches(const std::optional<bool> &ask, const std::optional<bool> &have) {
+  return !ask.has_value() || !have.has_value() || ask.value() == have.value();
+}
+
+// Similar to OptionalMatches, but for JobID or ActorID.
+template <typename IdType>
+bool IdMatches(const IdType &ask, const IdType &have) {
+  return ask.IsNil() || have.IsNil() || ask == have;
+}
```

Comment:
Can be done in a follow-up PR. For consistency, we probably should use `std::Optional<IdType>` as well instead of relying on `IdType::Nil`.

Reply:
added comment